### DISCUSSION TITLE
Migrating from Mydumper 0.10 to 0.14 - metadata file not recognizable

### DISCUSSION THREAD

**Discussion by viswaprsnl** [mydumper/mydumper#1238] [Q&A] [answered]
New user here . I am trying to migrate from .10 version to .14 version and i see my imports getting failed due to metadata file not in format .Time being i am deleting the metadata file and creating a new empty file .I wanted to see if there is any other option to make this straight .

Also do we need the same versions of loader and dumper to successfully remediate this issue ?

**Comment by aqueos** [comment, marked as answer]
Hi,

  This seems to be a major trouble with mydumper as, as i understand it, you need the same version that did the dump to restore it.

  Cross version is not supported (can work but..). This means that a long term backup is an issue because you have to keep track of the version and keep the .deb/.rpm of the version you used just in case.

  As you cannot install several versions of mydumper at the same time, it can be a challenge to manage if you recover a backup from one or more years ago.

 Especially if your linux  distrib version is now abandoned you have to be sure to have the mydumper .deb hanging around in case it disappear from a repo somewhere :)

 Sometimes it can work but you need to have the package used for each dump to be sure you will be able to recover them later.
 As emergency mesure i think recovering individual sql file should work.

  I did not knowed that removing the metadatafile helped cross versions :)


best regards,
Ghislain.

**Reply by viswaprsnl** [reply]
@aqueos what is the rationale behind mydumper .10 version not getting supported in centos 9? is there any plan /road map for this to get solved ?

**Reply by davidducos** [reply]
@viswaprsnl newer version of MyDumper are supported in centos 9. Take into account that MyDumper v0.10 is an old version, we already at v0.15.

**Reply by aqueos** [reply]
support is mainly a manpower issue, funding is not here to support LTS like release i suppose as it would need a person dedicated to it so to not steal dev time from the project.

david, is myloader capable of working with old version dumps if we remove metadata file or replace it with an empty one ? 

in any case this is an issue that we must work around, here i keep the .deb of old version to reinstall them if needed.

**Reply by davidducos** [reply]
This is a good opportunity to document this:

How MyDumper decides to change major version upgrade, minor version, patching? 

First of all, MyDumper is v0.A.B-C , that ZERO (0) at the begining means that we still developing MyDumper, as I usually say, MyDumper can be used for production but it is not production ready. The goal that I set every year is to release v1 which is going to be the first stable release and production ready. When is going to happen? I'm not going to answer now, but check below!

So, during v0, A means major version, B means minor and C is patching. 
When we change A?
If I found a backward compatibility at file level, then I increase A. This means that file manipulation will not the same on different major versions. It doesn't means that it will never work, there might be scenarios where it could work, but I'm not documenting them as the amount of possibilities might be huge.

When we change B?
We try to fill a bucket as much as possible with features and bug fixes, and after a couple of months or when we found bugs that can't be patch to current version, we release a new version.

When we change C?
As soon as we release a major or minor version, users start to detect bugs. Mostly are easy fixes that we can patch and release a patching version. Usually patching version are releases during the first 2 weeks after a major/minor release.

When are we going to release a v1 / Production Ready?
We have been working to understand all the use cases that we have. However, there are new use cases all the time and it has been challenging to change mydumper and myloader to work as one single tool. What I can say right now, if we don't change or find something that we need to change in the main engine of mydumper and myloader for at least 6 months, then we are going to be able to release a v1. Honestly, I hope to start 2024 with a stable production ready version, as last changes that we did on mydumper and myloader opened the engines and reduced the memory usage which was a huge issue before.

**Reply by davidducos** [reply]
https://github.com/mydumper/mydumper/wiki/Versioning
